Code/RDGeneral/Dict.h

```diff
@@ -167,14 +175,66 @@ class Dict {
   */
   template <typename T>
   void setVal(const std::string &what, T &val) {
-    _data[what] = toany(val);
-  };
-  //! \overload
-  template <typename T>
-  void setVal(const char *what, T &val) {
-    std::string key = what;
-    setVal(key, val);
+    _hasNonPodData = true;
+    for(size_t i=0; i< _data.size(); ++i) {
+      if (_data[i].key == what) {
+        _data[i].val = val;
+        return;
+      }
+    }
+    _data.push_back(Pair(what, val));
   };
+
```

Comment:
Lots of duplicate code in here. How about adding a private method like this:
```
  template <typename T>
  void setValHelper(const std::string &what, const T &val, bool isPOD = true) {
    if (!isPOD) _hasNonPodData = true;
    for (size_t i = 0; i < _data.size(); ++i) {
      if (_data[i].key == what) {
        _data[i].val = val;
        return;
      }
    }
    _data.push_back(Pair(what, val));
  };
```
and calling that from the other versions of setVal?